PxGraf/Models/Metadata/DimensionExtensions.cs

```diff
@@ -0,0 +1,103 @@
+﻿#nullable enable
+using Px.Utils.Language;
+using Px.Utils.Models.Metadata.Dimensions;
+using Px.Utils.Models.Metadata.MetaProperties;
+using Px.Utils.Models.Metadata;
+using PxGraf.Data.MetaData;
+using PxGraf.Models.Queries;
+using PxGraf.Utility;
+using System.Collections.Generic;
+using System.Linq;
+using System;
+using Px.Utils.Models.Metadata.Enums;
+
+namespace PxGraf.Models.Metadata
+{
+    /// <summary>
+    /// Extension methods for <see cref="IReadOnlyDimension"/> objects.
+    /// </summary>
+    public static class DimensionExtensions
+    {
+        /// <summary>
+        /// Converts the given <see cref="IReadOnlyDimension"/> object to a <see cref="Variable"/> object.
+        /// </summary>
+        /// <param name="dimension">The <see cref="IReadOnlyDimension"/> object to convert.</param>
+        /// <param name="dimensionQueries">The dimension queries to use for the conversion.</param>"
+        /// <param name="meta">The matrix metadata to append missing content value information from.</param>
+        /// <returns>A <see cref="Variable"/> object created from the given <see cref="IReadOnlyDimension"/> object.</returns>
+        public static Variable ConvertToVariable(this Dimension dimension, Dictionary<string, DimensionQuery> dimensionQueries, IReadOnlyMatrixMetadata meta)
```

Comment:
Since this method takes the entire meta and the queries as parameter, maybe this should not be an extension of Dimension?